client/src/components/FontAwesomeIcon.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './FontAwesomeIcon.module.css';
+
+/**
+ * @typedef {import('@fortawesome/fontawesome-common-types').IconDefinition} IconDefinition
+ */
+
+/**
+ * @typedef {object} FontAwesomeIconProps
+ * @property {IconDefinition} icon Icon
+ */
+
+/**
+ * Lightweight FontAwesome icon component
+ *
+ * @param {FontAwesomeIconProps} props Props
+ * @returns {React.Component<FontAwesomeIconProps>} Component
+ */
+export default function FontAwesomeIcon({ icon }) {
+  const [width, height] = icon.icon;
+  const svgPathData = icon.icon[4];
```

Comment:
I feel like this will break with other icons? Pls confirm?

Reply:
This will work with all fontawesome icons through NPM. Their data structure is just weird. I rolled my own component since the official react bindings include 30kB of polyfills.

Reply:
Yeah that's fair enough. Totally understand why you did this. Leave a comment  somehow referring to this problem? I think another developer reading this line will be confused here.